src/java/org/apache/cassandra/cql3/functions/FunctionParameter.java

```diff
@@ -336,4 +341,55 @@ public String toString()
             }
         };
     }
+
+    /**
+     * @param type the type of the vector elements
+     * @return a function parameter definition that accepts values of type {@link VectorType} with elements of the
+     * specified {@code type} and any dimensions.
+     */
+    static FunctionParameter vector(CQL3Type type)
+    {
+        return new FunctionParameter()
+        {
+            @Override
+            public AbstractType<?> inferType(String keyspace,
+                                             AssignmentTestable arg,
+                                             @Nullable AbstractType<?> receiverType,
+                                             List<AbstractType<?>> inferredTypes)
```

Comment:
do we need a  annotation  of nullable for infferredType as line 60 ?

Reply:
We do, added